packages/server/src/question/question.subscriber.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  EventSubscriber,
+  EntitySubscriberInterface,
+  Connection,
+  UpdateEvent,
+} from 'typeorm';
+import {
+  NotificationService,
+  NotifMsgs,
+} from '../notification/notification.service';
+import { QuestionModel } from './question.entity';
+import { ClosedQuestionStatus } from '@template/common';
+
+@EventSubscriber()
+export class QuestionSubscriber
+  implements EntitySubscriberInterface<QuestionModel> {
+  notifService: NotificationService;
+  constructor(connection: Connection, notifService: NotificationService) {
+    this.notifService = notifService;
+    connection.subscribers.push(this);
+  }
+
+  listenTo() {
+    return QuestionModel;
+  }
+
+  // Notify students when they get near the top of the queue
+  async afterUpdate(event: UpdateEvent<QuestionModel>) {
+    // if status updated to closed
```

Comment:
Looks like this is triggered when any question's status is set to a `ClosedQuestionStatus`. Is it possible that if a question below the first three questions is changed to a `ClosedQuestionStatus` then the person in third place will get notified multiple times? (e.g when a student deletes their question or the TA removes it from the queue)

Reply:
good catch. Crap. not sure how to fix

Reply:
fixed now by checking that the student wasn't already third in the queue

Reply:
something to keep in mind is that if there are three TAs helping students, and we're only checking for open questions to notify, no one's gonna get notified.